google/datalab/contrib/pipeline/_pipeline.py

```diff
@@ -247,20 +253,21 @@ def _get_query_parameters(input_query_parameters):
     Returns:
       Validated object containing query parameters.
     """
-    jsonschema.validate({'parameters': input_query_parameters},
-                        google.datalab.bigquery.commands._bigquery.query_params_schema)
-
     parsed_params = []
-    for param in input_query_parameters:
-      parsed_params.append({
-        'name': param['name'],
-        'parameterType': {
-          'type': param['type']
-        },
-        'parameterValue': {
-          'value': param['value']
-        }
-      })
+    if input_query_parameters:
+      jsonschema.validate({'parameters': input_query_parameters},
```

Comment:
I am not sure I have fully followed the logic here, but raising this as a potential issue to think through: validation should have already happened earlier at the command handling phase. Wouldn't that be better?